From 517f841ccac59b0579e706dd768d6c7b1a7e7552 Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Fri, 12 Mar 2021 10:37:21 -0800
Subject: [PATCH] Use asm/type.h for kernel types

This ensures that right headers for types is included otherwise it can
conflict for some platforms e.g. ppc64 where it includes the underlying
files conditionally

asm/types.h is

if !defined(__SANE_USERSPACE_TYPES__) && defined(__powerpc64__) && !defined(__KERNEL__)
 include <asm-generic/int-l64.h>
else
 include <asm-generic/int-ll64.h>
endif

Upstream-Status: Submitted [https://github.com/westerndigitalcorporation/ufs-utils/pull/29]
Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 ufs.h      | 1 -
 ufs_cmds.h | 2 +-
 ufs_ffu.c  | 1 -
 3 files changed, 1 insertion(+), 3 deletions(-)

diff --git a/ufs.h b/ufs.h
index dc51367..0321c54 100644
--- a/ufs.h
+++ b/ufs.h
@@ -4,7 +4,6 @@
 
 #ifndef UFS_H_
 #define UFS_H_
-#include <asm-generic/int-ll64.h>
 #include "ioctl.h"
 #include "scsi_bsg_util.h"
 
diff --git a/ufs_cmds.h b/ufs_cmds.h
index 1e7e24d..5c5b045 100644
--- a/ufs_cmds.h
+++ b/ufs_cmds.h
@@ -5,7 +5,7 @@
 #define UFS_CMNDS_H_
 
 #include "options.h"
-#include <asm-generic/int-ll64.h>
+#include <asm/types.h>
 
 
 enum field_width {
diff --git a/ufs_ffu.c b/ufs_ffu.c
index 2bcec44..99b479c 100644
--- a/ufs_ffu.c
+++ b/ufs_ffu.c
@@ -12,7 +12,6 @@
 #include <unistd.h>
 #include <stdint.h>
 #include <errno.h>
-#include <asm-generic/int-ll64.h>
 
 #include "ufs.h"
 #include "ufs_cmds.h"
-- 
2.30.2

